### PR TITLE
League stats offense

### PR DESCRIPTION
**Pull request by ckccameron**
Writes methods for best_offense and worst_offense. Also split up into helper methods; all methods tested for

### CODE REVIEW

File: lib/league_stats.rb [hunk 1]
```diff
@@ -17,11 +17,46 @@ def count_of_teams
...
+    end
+  end
+
+  def total_goals_by_team_id(team_id)
+    games_sorted_by_team_id(team_id).sum do |game_team|
+      game_team.goals.to_i
+    end
+  end
+
+  def average_goals_by_team_id(team_id)
```
Comment (IamNorma) [inline lib/league_stats.rb:41]: This is an opportunity for a module

File: test/league_stats_test.rb [hunk 2]
```diff
@@ -37,8 +37,30 @@ def test_it_can_count_teams
...
+    assert_equal 33, @league_stats.total_goals_by_team_id(5)
+    assert_equal 12, @league_stats.total_goals_by_team_id(24)
+  end
+
+  def test_it_can_average_goals_by_team_id
+    assert_equal 1.86, @league_stats.average_goals_by_team_id(1)
+    assert_equal 1.93, @league_stats.average_goals_by_team_id(18)
+  end
+
+  def test_it_can_find_best_offense
```
Comment (dhalverson) [inline test/league_stats_test.rb:59]: These tests look good, I think we can add some helper methods for the next set of Stats to reduce the time for each method.

File: lib/league_stats.rb [hunk 3]
```diff
@@ -17,11 +17,46 @@ def count_of_teams
...
+  def unique_team_ids
+    ids = @game_teams_collection.game_teams.each do |game_team|
+      game_team.team_id
+    end
+    ids.map do |game_team|
+      game_team.team_id.to_i
+    end.uniq
+  end
 
+  def games_sorted_by_team_id(team_id)
```
Comment (IamNorma) [inline lib/league_stats.rb:29]: great use of helper methods

File: lib/team.rb [hunk 4]
```diff
@@ -1,16 +1,16 @@
 class Team
 
   attr_reader :team_id,
-              :franchise_id,
-              :team_name,
+              :franchiseid,
```
Comment (dhalverson) [inline lib/team.rb:4]: Good catch here, we'll have to run through the rest of our attrs to make sure this isn't happening anywhere else

**Review by dhalverson — APPROVED** [review]
These all look good, Norma and I will get our methods added shortly for your review!

**Pull request merged**